tests/utils/syscall_handler.py

```diff
@@ -262,6 +264,25 @@ def storage_write(self, segments, syscall_ptr):
             value=segments.memory[syscall_ptr + 2],
         )
 
+    def replace_class(self, segments, syscall_ptr):
+        """
+        Record the replaced class hash in the internal mock object and update the class_hash attribute.
+
+        Syscall structure is:
+
+            struct ReplaceClass {
+                selector: felt,
+                class_hash: felt,
+            }
+        """
+        class_hash = segments.memory[syscall_ptr + 1]
+        address = self.contract_address
+        self.mock_replace_class(
+            address=address,
+            class_hash=class_hash,
+        )
+        self.class_hash = class_hash
```

Comment:
why would you want to do this? + you are not updating here the class attribute, but an instance one, that's why the test doesn't pass

Reply:
but it doesn't make sense indeed